app/app/static/custom/ourscript.js

```diff
@@ -306,8 +312,26 @@ $('#librariansAvailableIncrement').click(function () {
 	document.getElementById('librariansAvailableCount').innerHTML = next;
 });
 
+function genChartsForCurrentWeek() {
+	// Get current date
+	var date = new Date();
+
+	// Gets the most recently passed Sunday and the next upcomming Saturday
+	var firstDay = new Date(date.setDate(date.getDate() - date.getDay())).toJSON().slice(0,10);
+	var lastDay = new Date(date.setDate(date.getDate() - date.getDay() + 6)).toJSON().slice(0,10);
+
+	// Format the dates to the appropriate dateTime format
+	const startTime = `${firstDay} 00:00:00.0000000`;
+	const endTime = `${lastDay} 23:59:59.999999`;
+	// Send html request
+	createDataChart({
+		startTime,
+		endTime
+	})
+}
```

Comment:
Can use default parameter to prevent repetition code in 554:570
``` 
function genChartsForGivenWeek(date = new Date()) {
// Gets the most recently passed Sunday and the next upcomming Saturday
	var firstDay = new Date(date.setDate(date.getDate() - date.getDay())).toJSON().slice(0,10);
	var lastDay = new Date(date.setDate(date.getDate() - date.getDay() + 6)).toJSON().slice(0,10);
....
}